compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
Thanks for the feedback, @andrewcraik @mstoodle 

> I am not sure that is enough... the optimizer dispatch loop checks a regex as well as the isDisabled call (see line 1624 onward in OMROptimizer.cpp).

> If we are going to standardize this logic then optimizer invocation sequence should be using this same query for its early exit at the point indicated.

The regex logic from line 1624 onward in OMROptimizer.cpp is currently **only** used in the optimizer dispatch loop and all the other consumers today (mostly other optimizations) are using comp()->isDisabled(optNum) directly. So adding this logic to the isDisabled/isEnabled method will potentially change the current behavior in a number of places.

> Further rather than always writing optimizer()->isEnabled.... wouldn't it make sense for the root Optimization to gain an enabled() or disabled() function that wraps up this logic so it is easy to redirect the handling again if desired in the future?

We do have an enabled() query on the root OptimizationManager class today. The optimizer()->isEnabled method was added as a wrapper for the underlying enabled() method as a convenience.

So another option that just occurred to me is that the isEnabled query on optimizer can contain the regex logic and the enabled query on optimization doesn't. This allows consumers to pick and choose what kind of checking they want to perform. However, this might generate more confusion as two seemingly similar queries actually do different checking under the hood?